.github/workflows/upload-deb-to-release.yaml

```diff
@@ -0,0 +1,188 @@
+              echo "Downloaded RELEASE and signed RELEASE files successfully."
+              break
+            else
+              RETRY_COUNT=$((RETRY_COUNT+1))
+              echo "Retrying in $RETRY_SLEEP_SECONDS seconds... (Attempt $RETRY_COUNT of $MAX_RETRIES)"
+              sleep $RETRY_SLEEP_SECONDS
+            fi
+          done
+
+          if [ $RETRY_COUNT -eq $MAX_RETRIES ]; then
+            echo "Max retries reached. Exiting..."
+            exit 1
+          fi
+
+          mv RELEASE.gpg dists/noble/
+          mv RELEASE dists/noble/
+
+          sudo gpg --import pool/main/f/run-finch/publickey.pem && sudo gpg --export --armor EC2A8C75A0F2C15B4B606642EC6C05BFDC5DA32C > GPG_KEY.pub
+          
+          # TEMP
+          cd pool
+          tree
+          cd ../dists
+          tree
+          cd ../
+      - name: Configure Artifacts AWS credentials
+        uses: aws-actions/configure-aws-credentials@b47578312673ae6fa5b5096b330d9fbac3d116df # v4.2.1
+        with:
+          role-to-assume: ${{ secrets.ROLE }}
+          role-session-name: update-latest-version-in-s3
+          aws-region: ${{ secrets.REGION }}
+      - name: Upload repository files to repository bucket
+        run: |
+          aws s3 cp pool/main/f/run-finch/run-finch_${{ needs.get-tag-and-version.outputs.version }}_amd64.deb s3://${{ secrets.ARTIFACT_BUCKET_NAME }}/pool/main/f/run-finch/
+          aws s3 cp pool/main/f/run-finch/run-finch_${{ needs.get-tag-and-version.outputs.version }}_amd64.deb.sig s3://${{ secrets.ARTIFACT_BUCKET_NAME }}/pool/main/f/run-finch/
+          aws s3 cp pool/main/f/run-finch/run-finch_${{ needs.get-tag-and-version.outputs.version }}_arm64.deb s3://${{ secrets.ARTIFACT_BUCKET_NAME }}/pool/main/f/run-finch/
+          aws s3 cp pool/main/f/run-finch/run-finch_${{ needs.get-tag-and-version.outputs.version }}_arm64.deb.sig s3://${{ secrets.ARTIFACT_BUCKET_NAME }}/pool/main/f/run-finch/
+          aws s3 cp pool/main/f/run-finch/publickey.pem s3://${{ secrets.ARTIFACT_BUCKET_NAME }}/pool/main/f/run-finch/
+
+          aws s3 cp dists/noble/RELEASE s3://${{ secrets.ARTIFACT_BUCKET_NAME }}/dists/noble/
+          aws s3 cp dists/noble/RELEASE.gpg s3://${{ secrets.ARTIFACT_BUCKET_NAME }}/dists/noble/
+          aws s3 cp dists/noble/main/binary-amd64/Packages s3://${{ secrets.ARTIFACT_BUCKET_NAME }}/dists/noble/main/binary-amd64/
+          aws s3 cp dists/noble/main/binary-amd64/Packages.gz s3://${{ secrets.ARTIFACT_BUCKET_NAME }}/dists/noble/main/binary-amd64/
+          aws s3 cp dists/noble/main/binary-arm64/Packages s3://${{ secrets.ARTIFACT_BUCKET_NAME }}/dists/noble/main/binary-arm64/
+          aws s3 cp dists/noble/main/binary-arm64/Packages.gz s3://${{ secrets.ARTIFACT_BUCKET_NAME }}/dists/noble/main/binary-arm64/
+
+          aws s3 cp GPG_KEY.pub s3://${{ secrets.ARTIFACT_BUCKET_NAME }}/
```

Comment:
Shouldn't we be using a /deb sub-directory here as to not clutter the artifact repo?